sklearn/naive_bayes.py

```diff
@@ -512,8 +512,40 @@ class _BaseDiscreteNB(_BaseNB):
 
     __init__
     _joint_log_likelihood(X) as per _BaseNB
+    _update_feature_log_prob(alpha)
+    _count(X, Y)
     """
 
+    @abstractmethod
+    def _count(self, X, Y):
+        """Update counts that are used to calculate probabilities.
+
+        The counts make up a sufficient statistic extracted from the data.
+        Accordingly, this method is called each time ``fit`` or ``partial_fit``
+        update the model. The number and composition of counts depend on a
+        concrete model, but ``self.class_count_`` must be updated in any case.
```

Comment:
The fact that this is an abstract method, means that `_count` depends on the concrete model. I think it's better to be direct and say that `self.class_count_` must be updated here:

```suggestion
        update the model. `self.class_count_` must be updated in this method.
```

(Since the `_BaseDiscreteNB._update_class_log_prior` can require it)

Reply:
Yes, it goes without saying that `_count` depends on a concrete model, since it is an abstract method. My point is that in addition to mandatory `self.class_count_`, other counts may be present, for example, `self.feature_count_` and `self.feature_all_` in `ComplementNB`. If you remove 

> The number and composition of counts depend on a concrete model

then it is easy to think that updating `self.class_count_` is the only purpose of method `_count`, which is not the case.

Reply:
The "but" in the current sentence is confusing to me. How about:

```suggestion
        update the model. `class_count_` and `feature_count_` must be updated
        here along with any model specific counts. 
```

This tells the subclass to update `class_count_` and `feature_count_`, but does not limit them to only those attributes.

(I see that `_BaseDiscreteNB._init_counters` does create the counts, so it makes sense to include them in the docstring.)

Reply:
I see. Yes, I like this last phrasing too and that `feature_count_` is mentioned.